lib/rom/rails/model/validator/uniqueness_validator.rb

```diff
@@ -66,8 +68,8 @@ def rom
         # @return [TrueClass,FalseClass]
         #
         # @api private
-        def unique?(name, value)
-          relation.unique?(name => value)
+        def unique?(name, value, scope)
+          relation.where(scope).unique?(name => value)
```

Comment:
OK :) To clarify - `Relation#unique?` is an interface that was introduced specifically for uniqueness validation. Currently only rom-sql implements it but others can follow :)